### PR TITLE
ScalaTest deps to toolchain

### PR DESCRIPTION
**Pull request by liucijus**
ScalaTest deps on toolchain, part of #940

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -38,8 +38,51 @@ declare_deps_provider(
 `junit_classpath_provider` (deps_id `junit_classpath`) is where classpath required for junit tests
 is defined.
 
-Toolchain must be registerd in your `WORKSPACE` file: 
+ScalaTest support can be enabled by configuring a provider with an id `scalatest_classpath`:
```
Comment (ittaiz) [inline docs/testing.md:41]: unrelated to this change but this doc seems to suggest that in order to use scalatest I need to configure it. Why doesn't it show the easy defaults way?

**Review by ittaiz — APPROVED** [review]

File: docs/testing.md (hunk 1 above)
Comment (liucijus) [reply docs/testing.md:41]: Not sure I understood your comment, there are no defaults when setting up custom toolchain

Comment (ittaiz) [reply docs/testing.md:41]: Is `docs/testing.md` only about custom testing toolchains?

Comment (liucijus) [reply docs/testing.md:41]: Probably not, but I think it requires a lot more effort to rethink/rewrite whole setting up rules scala docs. I think it should be done outside this PR.

Comment (ittaiz) [reply docs/testing.md:41]: My point- configuring custom deps became more complicated (but more powerful and we're ok with it).
If the readme before caused people to think they need to do X to use it now it seems like one needs to do 5X.
Can you please add a note here clarifying that this is needed only for the custom case?

Comment (liucijus) [reply docs/testing.md:55]: I have reorganized doc and added a note about custom deps

Comment (ittaiz) [reply docs/testing.md:55]: Thanks. I appreciate it

**Review by ittaiz — APPROVED** [review]

File: testing/scalatest.bzl [hunk 2]
```diff
@@ -0,0 +1,6 @@
+def scalatest_repositories():
+    # currently ScalaTest dependencies are already loaded via //scala:scala.bzl#scala_repositories()
+    pass
+
+def scalatest_toolchain():
+    native.register_toolchain("//testing:scalatest_toolchain")
```
Comment (TheKaur) [inline testing/scalatest.bzl:6]: @liucijus Is this supposed to be `register_toolchains`? 
Added the following in my WORKSPACE file:
```
load("@io_bazel_rules_scala//testing:scalatest.bzl", "scalatest_toolchain")

scalatest_toolchain()
```
and it errored out with:
```
Error: no native function or rule 'register_toolchain'
```

Comment (liucijus) [reply testing/scalatest.bzl:6]: Thanks for reporting, here's the fix: https://github.com/bazelbuild/rules_scala/pull/1130

**Pull request merged**